src/db_adapters/receipts.rs

```diff
@@ -17,10 +17,98 @@ pub(crate) async fn store_receipts(
     pool: &Pool<ConnectionManager<PgConnection>>,
     receipts: Vec<&near_indexer::near_primitives::views::ReceiptView>,
     block_hash: &str,
+    no_strict_mode: bool,
 ) {
     let mut skipping_receipt_ids =
         std::collections::HashSet::<near_indexer::near_primitives::hash::CryptoHash>::new();
 
+    let tx_hashes_for_receipts = find_tx_hashes_for_receipts(
+        &pool,
+        receipts.iter().map(|r| r.receipt_id.to_string()).collect(),
+    )
+    .await;
+
+    let mut receipt_models: Vec<models::receipts::Receipt> = receipts
+        .iter()
+        .filter_map(|r| {
+            if let Some(transaction_hash) =
+                tx_hashes_for_receipts.get(r.receipt_id.to_string().as_str())
+            {
+                Some(models::Receipt::from_receipt_view(
+                    r,
+                    block_hash,
+                    transaction_hash,
+                ))
+            } else {
+                warn!(
+                    target: crate::INDEXER_FOR_EXPLORER,
+                    "Skipping Receipt {} as we can't find parent Transaction for it.",
+                    r.receipt_id.to_string()
+                );
+                skipping_receipt_ids.insert(r.receipt_id);
+                None
+            }
+        })
+        .collect();
+
+    if !no_strict_mode {
+        let receipt_models_ = loop {
+            let mut models: Vec<models::receipts::Receipt> = vec![];
+            let mut new_skipping_receipt_ids =
+                std::collections::HashSet::<near_indexer::near_primitives::hash::CryptoHash>::new();
+            if skipping_receipt_ids.is_empty() {
+                break models;
```

Comment:
Given we initiate the `models` vector on every iteration, it seems that it will always be empty here. Also, we need to make sure we don't mess up with the `index_in_chunk` here.